src/benchmark_runner.cc

```diff
@@ -289,23 +318,10 @@ BenchmarkRunner::IterationResults BenchmarkRunner::DoNIterations() {
   std::unique_ptr<internal::ThreadManager> manager;
   manager.reset(new internal::ThreadManager(b.threads()));
 
-  // Run all but one thread in separate threads
-  for (std::size_t ti = 0; ti < pool.size(); ++ti) {
-    pool[ti] = std::thread(&RunInThread, &b, iters, static_cast<int>(ti + 1),
-                           manager.get(), perf_counters_measurement_ptr,
-                           /*profiler_manager=*/nullptr);
-  }
-  // And run one thread here directly.
-  // (If we were asked to run just one thread, we don't create new threads.)
-  // Yes, we need to do this here *after* we start the separate threads.
-  RunInThread(&b, iters, 0, manager.get(), perf_counters_measurement_ptr,
-              /*profiler_manager=*/nullptr);
-
-  // The main thread has finished. Now let's wait for the other threads.
-  manager->WaitForAllThreads();
```

Comment:
I'm mildly uncomfortable that we are introducing non-obvious threading-related changes
in an otherwise straight-forward refactoring PR.
Would you be open to dropping this call in a preparatory PR?

Reply:
I can do this. Should I do a wider refactoring? E.g. I still don't understand this `// Acquire the measurements/counters from the manager, UNDER THE LOCK!` comment some lines below (and the lock itself). I have searched the blame for the root of this comment, but didn't find something until 2018. There was some mentioning of benchmarks running interleaved, but even then - all variables in question are owned by exactly one BenchmarkRunner.

Reply:
I think what we can "safely" do is:
1. Drop `ThreadManager::WaitForAllThreads`
2. Drop `ThreadManager::end_condition_`
3. Drop `ThreadManager::end_cond_mutex_`

I wouldn't touch anything else there, it does not seem nessesary for this change...